### ISSUE TITLE
File extension filter doesn't always work as expected

### ISSUE THREAD

**Issue by Sytten** [caido/caido#407]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help us know what to work on
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Bug description

A few users reported the extension filter not always working as expected. We will need to investigate to see if there is something wrong in the extractor or the filter itself.

**Comment by Sytten** [comment]
It might be that we keep the leading dot in the extension `.html` instead of `html`.
This was done to support things multi-dots extensions like `.tar.gz`, but we might want to change that behaviour.

**Comment by kmskrishna** [comment]
Yes. I was trying to filter `SVG` files like `!= svg`, which didn't work. But `!=.svg`worked fine.

In another case. I am trying to remove all CSS files.
There is a path like this `static/css/editProfileComponent.ba48b5ac.chunk.css`

I tried `!=css, !=.css, !=%.css, does not contain "%.css"` none worked. 

What worked is `does not match "%.css"` This was unexpected. I expected the previous ones to work.
